tendermint/src/types/proto/crypto.rs

```diff
@@ -56,6 +58,14 @@ impl TryFrom<inner::PublicKey> for PublicKey {
     }
 }
 
+impl FromStr for PublicKey {
```

Comment:
Not sure that this is a correct way of how `FromStr` should be implemented.

Reply:
The cosmos cli uses json based string to pass the key. And it's implementation for described case. 

Maybe, we can skip a part with key type declaration and hardcode `Ed25519`. Then we can read string with only key data.

Reply:
Okay. I get it now. You implemented `FromStr` 'cause of `clap`?

Reply:
I'll remind you that in our implementation cli command and command expected to be different structures, but which implement `TryFrom<T, Error = anyhow::Error>` and where I expected from user to do such conversions

Reply:
> Okay. I get it now. You implemented `FromStr` 'cause of `clap`?

I depends on input format. It is json string for now.

Reply:
> I'll remind you that in our implementation cli command and command expected to be different structures, but which implement TryFrom<T, Error = anyhow::Error> and where I expected from user to do such conversions

I had such conversion. I was asked to improve the cli with automatic conversion of `PublicKey` type.

Reply:
Why not use some type which wrapps string and tries to parse it into `PublicKey`?